src/fsharp/DependencyManager.Integration.fs

```diff
@@ -84,35 +84,46 @@ type ReflectionDependencyManagerProvider(theType: Type, nameProperty: PropertyIn
 
     static member InstanceMaker (theType: System.Type, outputDir: string option) =
         match ReflectionHelper.getAttributeNamed theType dependencyManagerAttributeName,
-                ReflectionHelper.getInstanceProperty<string> theType namePropertyName,
-                ReflectionHelper.getInstanceProperty<string> theType keyPropertyName,
-                ReflectionHelper.getInstanceMethod<bool * string list * string list> theType [| typeof<string>; typeof<string seq>; typeof<string> |] resolveDependenciesMethodName
-                with
-        | None, _, _, _ -> None
-        | _, None, _, _ -> None
-        | _, _, None, _ -> None
-        | _, _, _, None -> None
-        | Some _, Some nameProperty, Some keyProperty, Some resolveDependenciesMethod ->
-            Some (fun () -> new ReflectionDependencyManagerProvider(theType, nameProperty, keyProperty, resolveDependenciesMethod, outputDir) :> IDependencyManagerProvider)
+              ReflectionHelper.getInstanceProperty<string> theType namePropertyName,
+              ReflectionHelper.getInstanceProperty<string> theType keyPropertyName
+              with
+        | None, _, _
+        | _, None, _
+        | _, _, None -> None
+        | Some _, Some nameProperty, Some keyProperty ->
+            let resolveMethod = ReflectionHelper.getInstanceMethod<bool * string list * string list> theType [| typeof<string>; typeof<string>; typeof<string>; typeof<string seq>; typeof<string> |] resolveDependenciesMethodName
+            let resolveMethodEx = ReflectionHelper.getInstanceMethod<bool * string list * string list> theType [| typeof<string>; typeof<string seq>; typeof<string> |] resolveDependenciesMethodName
```

Comment:
this means there are two valid shapes for this method? 

maybe having a comment will clarify for the maintenance and the notebook integration can be shaped back to use the full signature?